BridgeSDK/BridgeAPI/SBBActivityManager.m

```diff
@@ -116,12 +116,18 @@ - (NSArray *)filterTasks:(NSArray *)tasks forDaysAhead:(NSInteger)daysAhead andD
 
 - (NSArray *)filterTasks:(NSArray *)tasks scheduledFrom:(NSDate *)startDate to:(NSDate *)endDate
 {
+    // It's in the date range [startDate, endDate) if:
+    //  - it never expires or expires on or after startDate AND
+    //  - it hasn't been marked finished or was marked finished on or after startDate AND
+    //  - it was scheduled before endDate
```

Comment:
This will return all (and only) scheduled activities that existed in a scheduled, but not finished or expired, state at any point during the requested span, following the logic of the server API. BridgeSDK doesn't know the app's intentions with the results (e.g. whether it's going to use them to show activities that are available and still need to be done or to show past activity results), so if for example the app wants all activities that were finished between a month ago and yesterday, it would need to get everything that existed in that time span and then filter out or ignore the ones that got finished after the end of yesterday, and a UI that wants to show available activities to do would need to filter out or ignore the ones already marked finished.